models/model_choice.py

```diff
@@ -93,12 +93,24 @@ def net(net_params, inference=False):
         model = coordconv.swap_coordconv_layers(model, centered=centered, normalized=normalized, noise=noise,
                                                 radius_channel=radius_channel, scale=scale)
 
-    if net_params['training']['state_dict_path']:
+    if not inference and net_params['training']['state_dict_path']:
+        assert Path(net_params['training']['state_dict_path']).is_file()
         state_dict_path = net_params['training']['state_dict_path']
         checkpoint = load_checkpoint(state_dict_path)
     elif inference:
         state_dict_path = net_params['inference']['state_dict_path']
+        assert Path(net_params['inference']['state_dict_path']).is_file(), f"Could not locate {net_params['inference']['state_dict_path']}"
         checkpoint = load_checkpoint(state_dict_path)
+    elif model_name == 'deeplabv3_resnet101':
+        # default to pretrained on coco (21 classes)
+        coco_model = models.segmentation.deeplabv3_resnet101(pretrained=True, progress=True,
+                                                        num_classes=21, aux_loss=None)
+        checkpoint = coco_model.state_dict()
+        temp_checkpoint = {}
+        temp_checkpoint['model'] = {k: v for k, v in checkpoint.items()}  # Place entire state_dict inside 'model' key
+        del coco_model
+        del checkpoint
+        checkpoint = temp_checkpoint
```

Comment:
Goal: load coco state_dict as a normal, user provided, state_dict.
Don't like having an elif statement for sure a specific case, but I couln't see any quick solutions otherwise.